### PR TITLE
populate sharepoint itemInfo

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

Currently, all drive backup and restore actions
populatet a details.OneDriveInfo struct.  This
change branches that struct between one-
drive and sharepoint info, depending on the
current source.

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* #1616

## Test Plan

- [x] :zap: Unit test

### CODE REVIEW

File: src/internal/connector/onedrive/item_test.go [hunk 1]
```diff
@@ -49,39 +51,53 @@ func TestItemIntegrationSuite(t *testing.T) {
...
+
+	odDrives, err := drives(ctx, suite, suite.user, OneDriveSource)
+	require.NoError(t, err)
 	// Test Requirement 1: Need a drive
-	require.Greaterf(suite.T(), len(drives), 0, "user %s does not have a drive", suite.user)
+	require.Greaterf(t, len(odDrives), 0, "user %s does not have a drive", suite.user)
 
 	// Pick the first drive
-	suite.driveID = *drives[0].GetId()
+	suite.userDriveID = *odDrives[0].GetId()
```
Comment (dadams39) [inline src/internal/connector/onedrive/item_test.go:92]: There exist inconsistent test based off picking the "first" Drive. Long-term do you see a case where arbitrarily picking a drive could cause a failure in the near future?

File: src/internal/connector/onedrive/item_test.go [hunk 2]
```diff
@@ -106,73 +122,90 @@ func (suite *ItemIntegrationSuite) TestItemReader() {
...
-	require.NotEmpty(suite.T(), itemInfo.ItemName)
+	require.NotNil(suite.T(), itemInfo.OneDrive)
+	require.NotEmpty(suite.T(), itemInfo.OneDrive.ItemName)
 
 	size, err := io.Copy(io.Discard, itemData)
 	require.NoError(suite.T(), err)
 	require.NotZero(suite.T(), size)
-	require.Equal(suite.T(), size, itemInfo.Size)
-	suite.T().Logf("Read %d bytes from file %s.", size, itemInfo.ItemName)
+	require.Equal(suite.T(), size, itemInfo.OneDrive.Size)
```
Comment (dadams39) [inline src/internal/connector/onedrive/item_test.go:138]: Unsure if this is the best metric. I believe @ashmrtn had a few issues where byte size within the store would vary. Not sure that a variance between these numbers would necessarily be an error.

**Review by dadams39 — APPROVED** [review]
Mostly nits: Please review prior to merging.

File: src/internal/connector/onedrive/item_test.go [hunk 3]
```diff
@@ -5,23 +5,25 @@ import (
...
 )
 
 type ItemIntegrationSuite struct {
 	suite.Suite
-	user    string
-	driveID string
-	client  *msgraphsdk.GraphServiceClient
-	adapter *msgraphsdk.GraphRequestAdapter
+	site        string
+	siteDriveID string
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/connector/onedrive/item_test.go:22]: Right you are.  I'll comment them out for now.

File: src/internal/connector/onedrive/item_test.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/connector/onedrive/item_test.go:92]: Possibly.  Depends on the conditions.  For example, user accounts should only have a single drive to begin with (their onedrive acct).  Sharepoint, on the other hand, may need additional specification.  Either way, those changes are out of the scope of this PR.

File: src/internal/connector/onedrive/item_test.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/internal/connector/onedrive/item_test.go:138]: The test has been consistently successful so far.  I don't see reason to change it.

**Pull request merged**